src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    /// <summary>
+    /// Provides JSON serialization-related metadata about a type.
+    /// </summary>
+    /// <typeparam name="T">The generic definition of the type.</typeparam>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class JsonTypeInfo<T> : JsonTypeInfo
+    {
+        internal JsonTypeInfo(Type type, JsonSerializerOptions options, ClassType classType) :
+            base(type, options, classType)
+        { }
+
+        internal JsonTypeInfo()
+        {
+            Debug.Assert(false, "This constructor should not be called.");
```

Comment:
Out of curiosity, why is the constructor needed at all?

Reply:
To prevent types outside of STJ.dll from deriving from this class.